src/NServiceBus.Core/UnitOfWork/TransactionScopes/TransactionScopeUnitOfWorkBehavior.cs

```diff
@@ -16,8 +16,7 @@ public HandlerTransactionScopeWrapperBehavior(TransactionOptions transactionOpti
         {
             if (Transaction.Current != null)
             {
-                await next().ConfigureAwait(false);
-                return;
+                throw new Exception("Ambient transaction already present, please remove the transaction scope unit of work by removing the call to .UnitOfWork().WrapHandlersInATransactionScope()");
```

Comment:
Fine, but we should not suggest which of these TS to remove. This is up to the user to decide.